### ISSUE TITLE
Mistake in error schema

### ISSUE THREAD

**Issue by volovyks** [near/nearcore#4025]
**Describe the bug** JSON RPC returns an error that does not exist in the [Error Schema.](https://github.com/near/nearcore/blob/master/chain/jsonrpc/res/rpc_errors_schema.json) Probably it's happening because of wrong naming. Error data from JSON RPC:

<pre>{
 TxExecutionError: {
 InvalidTxError: {
 ActionsValidation: {
 TotalPrepaidGasExceeded: 

 { total_prepaid_gas: 1000000000000000000, limit: 300000000000000, } 

,
 },
 },
 },
 }</pre>

[ActionsValidation](https://github.com/near/nearcore/blob/master/chain/jsonrpc/res/rpc_errors_schema.json#L717) in schema does not have any subtypes, so we can not determine the type of the error in `near-api-js` [walkSubtype](https://github.com/near/near-api-js/blob/master/src/utils/rpc_errors.ts#L47) function. On the other hand, there is [ActionsValidationError](https://github.com/near/nearcore/blob/master/chain/jsonrpc/res/rpc_errors_schema.json#L495) that is not linked anywhere. And it has a proper subtype.

It leads to problems like this: [https://github.com/near/near-api-js/issues/367](https://github.com/near/near-api-js/issues/367)

**To Reproduce** Check issue [https://github.com/near/near-api-js/issues/367](https://github.com/near/near-api-js/issues/367)

**Expected behavior** `TotalPrepaidGasExceeded` error should be a subtype of `ActionsValidation` error.

**Version (please complete the following information):**

*   nearcore commit: [https://github.com/near/nearcore/commit/3a31d76336bbc77d8d5633fd689f4cbcc4ef2bd4](https://github.com/near/nearcore/commit/3a31d76336bbc77d8d5633fd689f4cbcc4ef2bd4)

**Comment by frol** [comment]
Indeed, the relation between `ActionsValidation` and `ActionsValidationError`. We will address it on nearcore side.

Implementation details for future reference:

In order to make it backward-compatible, we will need to make an extension to the [RpcError macro](https://github.com/near/nearcore/blob/b973883a7fcf5a4534248c2bad89c9825bac019f/tools/rpctypegen/macro/src/lib.rs#L55-L63), so we can map `ActionsValidatorError` type back to the `ActionsValidation`.

https://github.com/near/nearcore/blob/b973883a7fcf5a4534248c2bad89c9825bac019f/core/primitives/src/errors.rs#L172-L175

I think we can add an attribute there:

```rust
#[derive(
    BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug, Clone, PartialEq, Eq, RpcError,
)]
#[rpc_error("ActionValidation")]
pub enum ActionsValidationError {
```

This requires some macro-magician work to `RpcError` (the attribute does not exist)

**Comment by volovyk-s** [comment]
@bowenwang1996 no, it's not fixed.

**Comment by trechriron** [comment]
@bowenwang1996  - This is blocking another issue on NEAR-API-JS. Who could we ask to pick this up? I appreciate your time!

**Comment by bowenwang1996** [comment]
@gmilescu could your team take a look at this?

**Comment by frol** [comment]
FYI, the quick fix, if you want it, is just rename `ActionsValidationError` to `ActionsValidation` in the JSON file, but it will get back to `ActionsValidationError` again on the next schema re-generation, which is why I proposed to extend the macro that generates the code.
